src/main/java/net/dv8tion/jda/core/requests/WebSocketClient.java

```diff
@@ -1006,32 +1008,43 @@ protected void handleEvent(JSONObject raw)
         }
     }
 
+    public boolean onBufferMessage(byte[] binary) throws IOException
+    {
+        if (binary.length >= 4 && getInt(binary, binary.length - 4) == ZLIB_SUFFIX)
```

Comment:
is the length is greater than 4, but ends in the ZLIB suffix, why do we just throw away the data that came before the suffix in this packet?